### PR TITLE
Pull request to allow multiple components of each type

### PR DESCRIPTION
**Pull request by OmniPotent42**
In the current version of the client, you have to pass a `components` object, with a 1:1 mapping between each component type. An example would be:

```
GoogleMapsClient.placesAutoComplete({
  input: '123 Main Street',
  types: ['address'],
  components: {
    country: 'us'
  }
})
```

Because you can only specify a string value for each component type, using the client it is impossible to select multiples of the same type, like 2 countries for example, a feature which has been possible to do through the API for over a year now.

My PR will allow you to do this:

```
GoogleMapsClient.placesAutoComplete({
  input: '123 Main Street',
  types: ['address'],
  components: {
    country: ['us', 'ca']
  }
})
```

Basically, you can now pass in an array for each component type instead of a string. The old way will still work as well.

### CODE REVIEW

**Comment by OmniPotent42** [comment]
I added the unit test in the `/spec/e2e` directory. Let me know if I did it wrong.

**Comment by stephenmcd** [comment]
The delay was merely due to not having a chance to look at this until now.

As for the change, I should have gone into more detail when I mentioned adding a test, sorry about that. We don't actually run the e2e tests anymore as the underlying data can change quite often which makes them brittle. What I should have mentioned is I think we just need a test that verifies adding single/multiple components params returns them as expected. There are probably some good examples in https://github.com/googlemaps/google-maps-services-js/blob/master/spec/unit/validate-spec.js (the new tests probably belong in there too).

Do you want to fix up the tests based on that?

**Comment by OmniPotent42** [comment]
I decided to create a new file for this test, since my changes were in the `convert` file and not really a part of the `validate` file, and I am directly testing the `pipedKeyValues` function.
